src/ngraph/op/broadcast.hpp

```diff
@@ -18,88 +18,169 @@
-
-            virtual std::shared_ptr<Node>
-                copy_with_new_args(const NodeVector& new_args) const override;
-
-            void infer_shape() override;
-            const AxisSet& get_initial_broadcast_axes() const { return m_initial_broadcast_axes; }
-            void set_initial_broadcast_axes(const AxisSet& initial_broadcast_axes)
+            class BroadcastLike : public v0::Broadcast
             {
-                m_initial_broadcast_axes = initial_broadcast_axes;
-            }
+            public:
+                NGRAPH_API
+                static constexpr NodeTypeInfo type_info{"BroadcastLike", 0};
+                const NodeTypeInfo& get_type_info() const override { return type_info; }
+                /// \brief Broadcast arg to the same shape as like_arg.
+                BroadcastLike() = default;
+                /// \brief Broadcast arg to the same shape as like_arg.
+                ///
+                /// Once the shape of like_arg is known, this op will be replaced with an equivalent
+                /// Broadcast op.
+                ///
+                /// \param arg The argument to be broadcast.
+                /// \param like_arg Provides the shape for the result.
+                /// \param initial_broadcast_axes indicates which axes will be broadcast. If empty,
+                ///        arg must be scalar and all axes are broadcast.
+                BroadcastLike(const Output<Node>& arg,
+                              const Output<Node>& like_arg,
+                              const AxisSet& initial_broadcast_axes);
+
+                virtual std::shared_ptr<Node>
+                    copy_with_new_args(const NodeVector& new_args) const override;
+
+                void infer_shape() override;
+                const AxisSet& get_initial_broadcast_axes() const
+                {
+                    return m_initial_broadcast_axes;
+                }
+                void set_initial_broadcast_axes(const AxisSet& initial_broadcast_axes)
+                {
+                    m_initial_broadcast_axes = initial_broadcast_axes;
+                }
+
+            protected:
+                AxisSet m_initial_broadcast_axes;
+            };
+        } // namespace v0
+
+        namespace v1
+        {
+            /// \brief Operation which "adds" axes to an input tensor, replicating elements from the
+            ///        input as needed along the new axes.
+            class Broadcast : public Op
+            {
+            public:
+                NGRAPH_API
+                static constexpr NodeTypeInfo type_info{"Broadcast", 1};
+                const NodeTypeInfo& get_type_info() const override { return type_info; }
+                /// \brief Constructs a broadcast operation.
+                Broadcast() = default;
+                /// \brief Constructs a broadcast operation.
+                ///
+                /// \param arg            Node that produces the input tensor to be broadcast.
```

Comment:
"Value" instead of "Node" since nodes can gave more than one output. Same for other node params.